### ISSUE TITLE
Can BansheeEngine integrate .Net Core in plan?

### ISSUE THREAD

**Issue by srxqds** [BearishSun/BansheeEngine#102]
If .Net Core can use on android and ios, it will integrate .net core in future?

**Comment by BearishSun** [comment]
It's unlikely, .NET core runtime has a very limited embedding interface and last time I spoke to those guys there didn't seem to be any plans to extend it. Mono on the other hand has an extremely expressive interface which Banshee currently relies on.

I'm interested what advantages do you think .NET Core has over Mono regarding gamedev?

**Comment by srxqds** [comment]
I think .net core can keep engine more simple and smaller assembly. 
And .net core should run more fast than mono because of the native implemention. 
I think use c# just like python but running more fast and developing more efficient. 
More embedding interface can provide by c++ layer.

**Comment by BearishSun** [comment]
The C++ layer is the embedding interface, and it's pretty much non-existent in .NET core. This means I'd have to emulate a lot of that functionality in C#, adding to cross-language calls which impacts performance, and adding to code complexity significantly.

That said, I am planning on adding a pure API wrapper layer for many different languages, including C#, Python, Lua and Java. The (C#) wrapper will not have tight integration with the runtime and will be able to run on all .NET platforms, including .NET Core. 

However this also means the script code will not be able to tightly integrate with the editor, so these API's will likely only be usable on their own (similar to MonoGame), while the editor will serve as content creation tool you can export scenes and data from (but not build directly from it).

**Comment by srxqds** [comment]
Yeah, it add more work for c++ layer,  but it also can more extensible. I can provide embedding interface by c++ layer，also can provide by script virtual machine (mono, python) itself. I can more freedom, more choice. 

By the way, I can't be waiting for the engine can run on macOS, hope it coming soon.

Thank you for your work.

**Comment by kingofoz** [comment]
I like lua. :-)

**Comment by ShalokShalom** [comment]
So I guess these guys have it easy since the engine itself is written in C#. While they seem to be very fine with the feature set. 

https://xenko.com/blog/toward-a-better-dotnet/

https://blogs.msdn.microsoft.com/dotnet/2016/09/26/introducing-net-standard/

See the Api ports link in the documentation: Its possible to check the code. 

Happy hacking

**Comment by realvictorprm** [comment]
@BearishSun can you link the discussion between you and the. NET Core devs?

**Comment by BearishSun** [comment]
I spoke to the CoreRT guys: https://github.com/dotnet/corert/issues/768

**Comment by realvictorprm** [comment]
Thank you very much :-)
